### PR TITLE
Arm backend: Move batchnorm decomposition to pass

### PR DESCRIPTION
**Pull request by oscarandersson8218**
### Summary
The decomposition logic of batchnorm is better suited for a pass than a node visitor.

Change-Id: I859912dc6ed437aa96a3bcdd819531dc8e02230e

cc @digantdesai @freddan80 @per @zingo

### CODE REVIEW

**Comment by zingo** [comment]
Please rebase now that  https://github.com/pytorch/executorch/commit/5f849ca02f61a42a22e53f25d3385bb7282fdec8  is merged.

**Review by zingo — APPROVED** [review]
